enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,256 @@
+also be possible to install LVMS and CSI components to a cluster that was initially deployed without them. This should
+not be a reversible process: uninstalling the rpms should not delete CSI components or cluster APIs from the cluster.
+Doing so endangers users' data and could lead to orphaning of LVM volumes. Instead, uninstalling the rpms will only
+ensure that LVMS and Snapshotting are not deployed on MicroShift startup/restart. It will be the user's responsibility
+to ensure there are not LVMS PVs in the cluster before deleting the LVMS and Snapshotting API instances.
+
+### Workflow Description
+
+**_Prerequisites_**
+
+- Host has LVM installed
+
+**_Installation with CSI Driver and Snapshotting_**
+
+1. User determines there is a requirement for persistent storage and volume snapshotting.
+2. User specifies an ostree blueprint which includes the following sections:
+    1. Packages: microshift, microshift-greenboot, microshift-networking, microshift-selinux, microshift-lvms, 
+       microshift-lvms-snapshotting
+    2. File (Optional): lvmd.yaml
+3. User compiles an ostree commit from the blueprint
+4. User deploys the ostree commit to host
+5. MicroShift host boots
+6. MicroShift starts
+7. MicroShift deploys:
+   1. LVMS CSI manifests
+   2. LVMS Snapshot manifests
+
+**_Installation without CSI Driver and Snapshotting_**
+
+1. User determines there is not a requirement for persistent storage and volume snapshotting.
+2. User specifies an ostree blueprint which includes the following sections:
+    1. Packages: microshift, microshift-greenboot, microshift-networking, microshift-selinux
+3. User compiles an ostree commit from the blueprint
+4. User deploys the ostree commit to host
+5. MicroShift host boots
+6. MicroShift starts
+
+**_Day-1|2 Installation_**
+
+1. User has previous installed MicroShift on a device.
+2. User later determines there is a requirement for persistent storage and volume snapshotting.
+3. User specifies an ostree blueprint which includes the following sections:
+    1. Packages:  microshift-lvms, microshift-lvms-snapshotting
+    2. File (Optional): lvmd.yaml
+4. User compiles an ostree commit from the blueprint
+5. User deploys the ostree commit to host
+6. MicroShift host reboots
+7. MicroShift starts
+8. MicroShift deploys:
+   1. LVMS CSI manifests
+   2. LVMS Snapshot manifests
+
+**Uninstallation**
+
+1. User has deployed a cluster with LVMS and CSI Snapshotting installed.
+2. User has run pods on the cluster with LVMS PVs.
+3. User decides to LVMS and snapshotting are no longer needed.
+4. User checks that there are no LVMS PVs, PVCs, or Snapshots in the cluster. If there are, it is the user's 
+   responsibility to prevent unintentional data loss or LVM volume orphaning before proceeding.
+5. User uninstalls microshift-lvms and microshift-lvms-snapshotting rpms.
+6. User deletes the relevant cluster API resources
```

Comment:
Should we swap steps 4 and 5? I mean, when is the data orphaned? 
Is it after RPM uninstall, or after the actions described in step 6?